src/types/package.go

```diff
@@ -72,3 +72,29 @@ type ZarfBuildData struct {
 	LastNonBreakingVersion     string            `json:"lastNonBreakingVersion,omitempty" jsonschema:"description=The minimum version of Zarf that does not have breaking package structure changes"`
 	Flavor                     string            `json:"flavor,omitempty" jsonschema:"description=The flavor of Zarf used to build this package"`
 }
+
+// PackageError is a struct that contains a finding about something wrong with a package
+type PackageError struct {
+	// YqPath is the path to the key where the error originated from, this is sometimes empty in the case of a general error
+	YqPath      string
+	Description string
+	// Item is the value of a key that is causing an error, for example a bad image name
+	Item string
+	// PackageNameOverride is used in the case of composition to let the user know which package the error originated from
+	// If it is not set the base package will be used when displaying the error
+	PackageNameOverride string
+	// PackageNameOverride is used in the case of composition to let the user know which package the error originated from
+	// If it is not set the base package will be used when displaying the error
+	PackagePathOverride string
+	Category            Severity
+}
+
+// Severity is the type of package error
+// Either Err or Warning
+type Severity int
+
+// different severities of package errors
+const (
+	SevErr Severity = iota + 1
+	SevWarn
+)
```

Comment:
These types and constants should probably live in the `src/pkg/packager/lint` package because they are used for linting and schema validation. It would move the data closer to the behavior and logic as well

Reply:
The plan is to have `packageError` used across Zarf as a way to store warnings or errors, as is done in #2585

Reply:
Which fields are used for linting/validation versus generic warnings/errors? Can we achieve better separation of concerns and clarity by having one struct for validation errors and another for generic warnings/errors? Rather than using the same struct for all errors and cases where the validation specific fields are not needed

Something like this:

```go
type PackageError struct {
	Description string
	Category    Severity
}

type ValidationErr struct {
	PackageError
	// Fields specific to ValidationErr
	YqPath               string
	Item                 string
	PackageNameOverride  string
	PackagePathOverride  string
}
```

Reply:
So the idea behind these fields is that packages outside of lint could create PackageErrors. Maybe finding would be a better name. I think in order to grab the context we want from errors we need creators of packageErrors to have access to these all of these fields even if they don't have something to put in all fields. 

`PackageNameOverride` & `packagePathOverride` are used when a error is found within an imported component so the user can be given the component that the error came from. 
'yqPath' is used when the error has a path in the code like `.metadata.name`
'Item' is used for additional context to the user, for example stating the image during an unpinned image warning